rasterio/_io.pyx

```diff
@@ -1745,18 +1746,19 @@ cdef class RasterUpdater(RasterReader):
 
         specifying a raster subset to write into.
         """
-        cdef void *hband
-        cdef void *hmask
-        if self._hds == NULL:
-            raise ValueError("can't write closed raster file")
-        hband = _gdal.GDALGetRasterBand(self._hds, 1)
-        if hband == NULL:
-            raise ValueError("NULL band mask")
-        if _gdal.GDALCreateMaskBand(hband, 0x02) != 0:
-            raise RuntimeError("Failed to create mask")
-        hmask = _gdal.GDALGetMaskBand(hband)
-        if hmask == NULL:
-            raise ValueError("NULL band mask")
+        cdef void *hband = NULL
+        cdef void *hmask = NULL
+
+        hband = self.band(1)
+        if GDALError.none != _gdal.GDALCreateMaskBand(hband, 0x02):
+            raise RasterioIOError("Failed to create mask.")
```

Comment:
I wasn't sure that `GDALCreateMaskBand()` called `CPLError()` but have confirmed that it does, so I can bring it inside the `CPLErrors` block.